src/synchro.rs

```diff
@@ -424,9 +432,11 @@ macro_rules! resampler_FftFixedout {
                     }
                 }
 
-                let mut wave_out = vec![Vec::new(); self.nbr_channels];
+                wave_out.resize(self.nbr_channels, Vec::new());
                 for chan in used_channels.iter() {
-                    wave_out[*chan] = self.output_buffers[*chan].clone();
+                    let output_buffer = self.output_buffers[*chan].len();
```

Comment:
Please rename this variable to something with "len".

Reply:
Will do, it's called `new_size` below.